### PR TITLE
Wearable UPS!

### PR DESCRIPTION
**Pull request by Rod995**
#### Summary
``SUMMARY: CONTENT "This makes a new Engineer's UPS, and makes all UPS types wearable."``
#### Purpose of change
Solves #25733 

#### Describe the solution
This makes a new Engineer's UPS, and makes all UPS types wearable. C++ was done by @Maskrow! 

#### Describe alternatives you've considered
Dunno, don't ask me.
Does need something to make it wearable with power armor tho.

### CODE REVIEW

File: data/json/items/tools.json [hunk 1]
```diff
@@ -90,7 +90,33 @@
...
+    "encumbrance" : 9,
+    "coverage" : 15,
+    "material_thickness" : 4,
+    "flags": [ "NO_UNLOAD", "RECHARGE", "BELTED", "UPS", "OVERSIZE" ]
+  },
+  {
+    "id": "mega_UPS_off",
+    "type": "TOOL_ARMOR",
+    "name": "Engineer's UPS",
+    "name_plural": "Engineers' UPS's",
```
Comment (Xhuis) [inline data/json/items/tools.json]: Names should be lowercased.

**Comment by FulcrumA** [comment]
Needs corrected summary as per the template.

**Comment by Mecares** [comment]
Maybe add a shoulder strap to the recipe?

**Comment by Marrim** [comment]
I stand by the thought the small ones should be worn at the hip, like the plastic canteen. Gives the player some choice. I guess torso still works if they resemble a messenger bag rather than a backpack.

Also, it looks like you changed the advanced UPS to have the energy efficiency of the normal UPS?

**Comment by Rod995** [comment]
Jenkins rebuild

**Comment by Rod995** [comment]
Should it be rarer than the Advanced UPS or nah @Marrim?

**Comment by Marrim** [comment]
Depends on where it would spawn.
It could be a very rare spawn at public works and power substations, but relatively common (more common than advanced UPS) among military gear.

File: src/visitable.cpp [hunk 2]
```diff
@@ -831,8 +826,16 @@ long visitable<Character>::charges_of( const std::string &what, long limit ) con
...
     if( what == "UPS" ) {
         long qty = 0;
-        qty = sum_no_wrap( qty, charges_of( "UPS_off" ) );
-        qty = sum_no_wrap( qty, long( charges_of( "adv_UPS_off" ) / 0.6 ) );
+        if( p && p->has_item_with_flag( "UPS" ) ) {
+            for( auto &i : p->all_items_with_flag( "UPS" ) ) {
+                qty = sum_no_wrap( qty, charges_of( i->typeId() ) );
+            }
+        }
+        if( p && p->has_item_with_flag( "ADV_UPS" ) ) {
```
Comment (BevapDin) [inline src/visitable.cpp]: You don't need to check whether there are any items with that flag, just iterate over them (as you do in the loop below). This is like
```C++
if( !some_container.empty() ) {
    for( elem : some_container ) {
        ...
    }
}
```
Loops handle empty ranges just fine. Same in the code above.

**Comment by Maskrow** [comment]
Thanks for the review! I’m still learning how to code. Will look into fixing it as soon as I have time.

File: src/iuse.cpp [hunk 3]
```diff
@@ -5317,7 +5317,7 @@ int iuse::toolmod_attach( player *p, item *it, bool, const tripoint & )
 
     auto filter = [&it]( const item & e ) {
         // don't allow ups battery mods on a UPS
-        if( it->has_flag( "USE_UPS" ) && ( e.typeId() == "UPS_off" || e.typeId() == "adv_UPS_off" ) ) {
+        if( it->has_flag( "USE_UPS" ) && ( e.has_flag ( "UPS" ) || e.has_flag( "ADV_UPS" ) ) ) {
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]: Astyle: `e.has_flag ( "UPS" )` should be `e.has_flag( "UPS" )`

File: src/visitable.cpp [hunk 4]
```diff
@@ -831,8 +826,16 @@ long visitable<Character>::charges_of( const std::string &what, long limit ) con
...
+            for( auto &i : p->all_items_with_flag( "ADV_UPS" ) ) {
+                qty = sum_no_wrap( qty, long( charges_of( i->typeId() ) / 0.6 ) );
+            }
+        }
```
Comment (ZhilkinSerg) [inline src/visitable.cpp]: Astyle: excessive whitespaces after curly brace

Comment (Rod995) [reply src/visitable.cpp]: Fixed, I think.

**Comment by Rod995** [comment]
Bah, this will get done later.

**Pull request closed without merging**